src/transport/SecureSessionTable.cpp

```diff
@@ -183,6 +239,109 @@ SecureSession * SecureSessionTable::EvictAndAllocate(uint16_t localSessionId, Se
+            bool doesBMatchSessionHintFabric =
+                b.mSession->GetPeer().GetFabricIndex() == evictionContext.GetSessionEvictionHint().GetFabricIndex();
+
+            //
+            // Sorting on Key2
+            //
+            if (doesAMatchSessionHintFabric != doesBMatchSessionHintFabric)
+            {
+                return doesAMatchSessionHintFabric > doesBMatchSessionHintFabric;
+            }
+
+            if (doesAMatchSessionHintFabric == doesBMatchSessionHintFabric)
+            {
+                //
+                // Sorting on Key3
+                //
+                if (a.mNumMatchingOnPeer != b.mNumMatchingOnPeer)
+                {
+                    return a.mNumMatchingOnPeer > b.mNumMatchingOnPeer;
+                }
+
+                if (a.mNumMatchingOnPeer == b.mNumMatchingOnPeer)
+                {
+                    int doesAMatchSessionHint = a.mSession->GetPeer() == evictionContext.GetSessionEvictionHint();
+                    int doesBMatchSessionHint = b.mSession->GetPeer() == evictionContext.GetSessionEvictionHint();
+
+                    //
+                    // Sorting on Key4
+                    //
+                    if (doesAMatchSessionHint != doesBMatchSessionHint)
+                    {
+                        return doesAMatchSessionHint > doesBMatchSessionHint;
+                    }
+
+                    if (doesAMatchSessionHint == doesBMatchSessionHint)
+                    {
+                        int aStateScore = 0, bStateScore = 0;
+
+                        auto assignStateScore = [](auto & score, const auto & session) {
+                            if (session.IsDefunct())
+                            {
+                                score = 2;
+                            }
+                            else if (session.IsActiveSession())
+                            {
+                                score = 1;
+                            }
+                            else
+                            {
+                                score = 0;
+                            }
+                        };
+
+                        assignStateScore(aStateScore, *a.mSession);
+                        assignStateScore(bStateScore, *b.mSession);
+
+                        //
+                        // Sorting on Key5
+                        //
+                        if (aStateScore != bStateScore)
+                        {
+                            return (aStateScore > bStateScore);
+                        }
+
+                        if (aStateScore == bStateScore)
+                        {
+                            //
+                            // Sorting on Key6
+                            //
+                            return (a->GetLastActivityTime() < b->GetLastActivityTime());
+                        }
+                    }
+                }
+            }
+        }
+
+        return false;
+    });
+
+#if 0
```

Comment:
I really prefer the single sort on multiple keys to multiple sorts.  Just remove this part.  Especially because that way we don't rely on sort stability, which makes it easier to change sort algorithms later if we want to.

Reply:
I had kept it in to poll folks as to which they preferred. Andrei preferred the single sort too (as do I), so will remove it.